server/src/main/java/org/opensearch/Version.java

```diff
@@ -239,13 +183,13 @@ private static Version fromStringSlow(String version) {
 
         try {
             final int rawMajor = Integer.parseInt(parts[0]);
-            if (rawMajor >= 5 && snapshot) { // we don't support snapshot as part of the version here anymore
-                throw new IllegalArgumentException("illegal version format - snapshots are only supported until version 2.x");
+            if (snapshot == true) { // we don't support snapshot as part of the version here anymore
+                throw new IllegalArgumentException("illegal version format - snapshot labels are not supported");
             }
-            if (rawMajor >=7 && parts.length == 4) { // we don't support qualifier as part of the version anymore
-                throw new IllegalArgumentException("illegal version format - qualifiers are only supported until version 6.x");
+            if (parts.length == 4) { // we don't support qualifier
```

Comment:
It seems the only valid number is 3, can we combine this with code at line 179?

Reply:
:+1:  agree there is some cleanup we can take care of in followup PR (e.g., move `is{Qualifier}` logic to LegacyESVersion, cleanup `toString`). I think we should also consider adding back support for optional build (e.g., rc1) which could be real useful for the plugins.